src/object/srv_obj.c

```diff
@@ -2397,9 +2397,8 @@ obj_inflight_io_check(struct ds_cont_child *child, uint32_t opc,
 	 * which otherwise might be written duplicately, which might cause
 	 * the failure in VOS.
 	 */
-	if ((flags & ORF_REBUILDING_IO) &&
-	    (!child->sc_pool->spc_pool->sp_disable_rebuild &&
-	      child->sc_pool->spc_rebuild_fence == 0)) {
+	if ((flags & ORF_REBUILDING_IO) && (ds_pool_rebuild_needed(child->sc_pool->spc_pool) &&
+					    child->sc_pool->spc_rebuild_fence == 0)) {
```

Comment:
[Not requesting changes] Strictly speaking, because we are reading `ds_pool.sp_self_heal` at least from an I/O xstream, and the same field may be written by `ds_pool_tgt_prop_update` from the system xstream, this is still a data race. We may need to revisit this in the future.

Reply:
Yes, @NiuYawei  told me that in Lustre and DAOS engineering efforts, it is common to assume that updating and fetch a uint64 value is atomic.